### PR TITLE
feat: add conversion codes to fit updated option key & update getInitialWidgetInheritOptions to fit updated spec

### PR DESCRIPTION
**Pull request by WANZARGEN**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci`, minor refactoring, etc.)
- [ ] Need discussion
- [ ] Not that difficult
- [ ] Approved feature branch merge to master

### Description
Please review in order by commits

### CODE REVIEW

File: apps/web/src/services/dashboards/widgets/_helpers/widget-inherit-options-helper.ts [hunk 1]
```diff
@@ -1,27 +1,47 @@
...
+        if (inheritanceMode === 'NONE') return;
+
+        if (!inheritanceMode || inheritanceMode === 'KEY_MATCHING') { // default inheritance mode is KEY_MATCHING
+            const variableKey = property.key;
+            const variableProperty = variablesSchema?.properties[variableKey];
+            if (!variableKey) return;
+            if (!variableProperty?.use) return;
+            refined[optionName] = { enabled: true, variable_key: variableKey };
             return;
         }
```
Comment (yuda110) [inline apps/web/src/services/dashboards/widgets/_helpers/widget-inherit-options-helper.ts:24]: praise: so clear

File: apps/web/src/services/dashboards/store/dashboard-detail-info.ts [hunk 2]
```diff
@@ -290,9 +290,11 @@ const getConvertedWidgetLayouts = (storedWidgetLayouts: DashboardModel['layouts'
 
     return storedWidgetLayouts.map((layout) => layout.map((widgetInfo) => {
         const convertedInheritOptions = getConvertedWidgetInheritOptions(widgetInfo.inherit_options);
+        const convertedWidgetOptions = getConvertedWidgetOptions(widgetInfo.widget_options);
```
Comment (yuda110) [inline apps/web/src/services/dashboards/store/dashboard-detail-info.ts:293]: nitpick: It seems like a NOTE is needed here too!

**Review by yuda110 — APPROVED** [review]
LGTM. so clear

File: apps/web/src/services/dashboards/store/dashboard-detail-info.ts (hunk 2 above)
Comment (WANZARGEN) [reply apps/web/src/services/dashboards/store/dashboard-detail-info.ts:293]: This is the function that is only used in this file.
I added a note to the code that uses this function, so I think it is unnecessary..! :)

**Pull request merged**